semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Using the Semantic Kernel Elasticsearch Vector Store connector (Preview)
+description: Contains information on how to use a Semantic Kernel Vector store connector to access and manipulate data in Elasticsearch.
+zone_pivot_groups: programming-languages
+author: flobernd
+ms.topic: conceptual
+ms.author: flobernd
+ms.date: 11/04/2024
+ms.service: semantic-kernel
+---
+# Using the Elasticsearch connector (Preview)
+
+> [!WARNING]
+> The Semantic Kernel Vector Store functionality is in preview, and improvements that require breaking changes may still occur in limited circumstances before release.
+
+## Overview
+
+The Elasticsearch Vector Store connector can be used to access and manage data in Elasticsearch. The connector has the following characteristics.
+
+| Feature Area                      | Support                                                                                                                          |
+|-----------------------------------|----------------------------------------------------------------------------------------------------------------------------------|
+| Collection maps to                | Elasticsearch index                                                                                                              |
+| Supported key property types      | string                                                                                                                           |
+| Supported data property types     | All types that are supported by System.Text.Json (etiher built-in or by using a custom converter)                                |
+| Supported vector property types   | <ul><li>ReadOnlyMemory\<float\></li><li>IEnumerable\<float\></li></ul>                                                           |
+| Supported index types             | <ul><li>HNSW (32, 8, or 4 bit)</li><li>FLAT (32, 8, or 4 bit)</li></ul>                                                          |
+| Supported distance functions      | <ul><li>CosineSimilarity</li><li>DotProductSimilarity</li><li>EuclideanDistance</li><li>MaxInnerProduct</li></ul>                |
```

Comment:
With the other connectors the names provided here are the same as what is supposed as a parameter.  E.g. to use `EuclideanDistance` you can pass "EuclideanDistance".  Is that the same here as well, for e.g. `MaxInnerProduct`?

Reply:
Thanks for the review!

Yes, that's the case here as well. The similarity measure / distance function is modelled as an enum in the underlaying Elasticsearch client library and I made sure to map the string constants defined in the SK abstractions accordingly. For `MaxInnerProduct`, which is specific to Elasticsearch, I've adapted the existing naming convention and made sure to as well map it to the corresponding internal enum member.

Reply:
Perfect, sounds good.  Thanks!